### ISSUE TITLE
Problem with Data Storage in Data Structure

### ISSUE THREAD

**Issue by riqs07** [riqs07/workBud#1]
stuff isnt updating in UI automatically as well as here is a issue in saving things in storage where it dosent always work and i dont think it is a the local storage level.

 i think the culprit is the findInData function because when it returns a folder to mutate it is giving me the folder in the projects part of the data structure not the current project or in local storage

 more investigation needed but now that i think can pinpoint the issue i should be able to solve it 

basically clean up how data is added to storage both in project and in data storage
essentially when something is updated in the current project/task make it equal to the same thing in the data structure and whenever the data structure is updated make it save to the local storage

def make that all one function so i dont have to keep repeating myself 

also prob going to make a function that makes a user input UI so i dont have to do document create every time

**Comment by riqs07** [comment]
fixed with updateData function

gets current task and finds it in the current project and makes it equal to one another
then use a pretty cool object.assign() to rewrite the version of the current project in the project structure then makes local storage equal to the project structure.
